### ISSUE TITLE
Besu randomly crashes and stops syncing

### ISSUE THREAD

**Issue by yuri-moens** [hyperledger/besu#5576]
**Expected behavior:** It works

**Actual behavior:** It doesn't work

**Frequency:** About 1-2 hours after restarting Besu

### Logs (if a bug)
```
2023-06-10 01:18:12.790+00:00 | vert.x-worker-thread-0 | INFO  | AbstractEngineNewPayload | Imported #17,446,465 / 127 tx / 16 ws / base fee 18.07 gwei / 8,465,311 (28.2%) gas / (0xe35c859e2798ebcc951e3743215dc69a0d60c04047c1c86bb646bdbbde47335b) in 0.600s. Peers: 50
2023-06-10 01:18:24.763+00:00 | vert.x-worker-thread-0 | INFO  | DefaultSynchronizer |
####################################################################################################
#                                                                                                  #
# Besu has identified a problem with its worldstate database.                                      #
# Your node will fetch the correct data from peers to repair the problem.                          #
# Starting the sync pipeline...                                                                    #
#                                                                                                  #
####################################################################################################
2023-06-10 01:18:24.765+00:00 | vert.x-worker-thread-0 | INFO  | TransactionPoolFactory | Disabling transaction handling during re-sync
2023-06-10 01:18:24.769+00:00 | vert.x-worker-thread-0 | WARN  | BonsaiWorldStateProvider | Invalid slot found for account 0xdfde78d2baec499fe18f2be74b6c287eed9511d7 at location 0x0c0c
2023-06-10 01:18:25.011+00:00 | vert.x-worker-thread-0 | INFO  | DefaultSynchronizer | Starting synchronizer.
2023-06-10 01:18:25.012+00:00 | vert.x-worker-thread-0 | INFO  | FastSyncDownloader | Starting sync
2023-06-10 01:18:25.058+00:00 | nioEventLoopGroup-3-5 | INFO  | SnapWorldStateDownloader | Downloading world state from peers for pivot block 17445216 (0x3643700ec320aa10c77861639c27734fd3f82364c028f527686720dac70b0635). State root 0x3eeffb63b373cab4a1f201e96f6d3a23b9ca27973860b68f1c3857055f669267 pending requests 0
2023-06-10 01:18:26.006+00:00 | nioEventLoopGroup-3-5 | INFO  | SnapWorldDownloadState | Running world state heal process from peers with pivot block 17445216
2023-06-10 01:18:33.145+00:00 | nioEventLoopGroup-3-4 | WARN  | DefaultBlockchain | Chain Reorganization +0 new / -1249 old
       Old - hash: 0xe35c859e2798ebcc951e3743215dc69a0d60c04047c1c86bb646bdbbde47335b, height: 17446465
       New - hash: 0x3643700ec320aa10c77861639c27734fd3f82364c028f527686720dac70b0635, height: 17445216
  Ancestor - hash: 0x3643700ec320aa10c77861639c27734fd3f82364c028f527686720dac70b0635, height: 17445216
2023-06-10 01:18:37.594+00:00 | vert.x-worker-thread-0 | INFO  | BonsaiWorldStateProvider | Archive rolling failed for block hash 0x3643700ec320aa10c77861639c27734fd3f82364c028f527686720dac70b0635
java.util.NoSuchElementException: No value present
        at java.base/java.util.Optional.get(Optional.java:143)
        at org.hyperledger.besu.ethereum.bonsai.BonsaiWorldStateProvider.rollMutableStateToBlockHash(BonsaiWorldStateProvider.java:208)
        at org.hyperledger.besu.ethereum.bonsai.BonsaiWorldStateProvider.lambda$getMutable$5(BonsaiWorldStateProvider.java:176)
        at java.base/java.util.Optional.flatMap(Optional.java:289)
        at org.hyperledger.besu.ethereum.bonsai.BonsaiWorldStateProvider.getMutable(BonsaiWorldStateProvider.java:174)
        at org.hyperledger.besu.ethereum.transaction.TransactionSimulator.getWorldState(TransactionSimulator.java:151)
        at org.hyperledger.besu.ethereum.transaction.TransactionSimulator.process(TransactionSimulator.java:131)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.EthCall.resultByBlockHeader(EthCall.java:78)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.AbstractBlockParameterOrBlockHashMethod.latestResult(AbstractBlockParameterOrBlockHashMethod.java:68)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.AbstractBlockParameterOrBlockHashMethod.handleParamTypes(AbstractBlockParameterOrBlockHashMethod.java:98)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.EthCall.response(EthCall.java:102)
        at org.hyperledger.besu.ethereum.api.jsonrpc.execution.BaseJsonRpcProcessor.process(BaseJsonRpcProcessor.java:44)
        at org.hyperledger.besu.ethereum.api.jsonrpc.execution.TracedJsonRpcProcessor.process(TracedJsonRpcProcessor.java:41)
        at org.hyperledger.besu.ethereum.api.jsonrpc.execution.TimedJsonRpcProcessor.process(TimedJsonRpcProcessor.java:45)
        at org.hyperledger.besu.ethereum.api.jsonrpc.execution.JsonRpcExecutor.execute(JsonRpcExecutor.java:92)
        at org.hyperledger.besu.ethereum.api.handlers.AbstractJsonRpcExecutor.executeRequest(AbstractJsonRpcExecutor.java:87)
        at org.hyperledger.besu.ethereum.api.handlers.JsonRpcObjectExecutor.execute(JsonRpcObjectExecutor.java:57)
        at org.hyperledger.besu.ethereum.api.handlers.JsonRpcExecutorHandler.lambda$handler$0(JsonRpcExecutorHandler.java:57)
        at java.base/java.util.Optional.ifPresentOrElse(Optional.java:196)
        at org.hyperledger.besu.ethereum.api.handlers.JsonRpcExecutorHandler.lambda$handler$2(JsonRpcExecutorHandler.java:54)
        at io.vertx.ext.web.impl.BlockingHandlerDecorator.lambda$handle$0(BlockingHandlerDecorator.java:48)
        at io.vertx.core.impl.ContextBase.lambda$null$0(ContextBase.java:137)
        at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:264)
        at io.vertx.core.impl.ContextBase.lambda$executeBlocking$1(ContextBase.java:135)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:833)
2023-06-10 01:18:37.605+00:00 | vert.x-worker-thread-14 | INFO  | BonsaiWorldStateProvider | Archive rolling failed for block hash 0x3643700ec320aa10c77861639c27734fd3f82364c028f527686720dac70b0635
java.util.NoSuchElementException: No value present
        at java.base/java.util.Optional.get(Optional.java:143)
        at org.hyperledger.besu.ethereum.bonsai.BonsaiWorldStateProvider.rollMutableStateToBlockHash(BonsaiWorldStateProvider.java:208)
        at org.hyperledger.besu.ethereum.bonsai.BonsaiWorldStateProvider.lambda$getMutable$5(BonsaiWorldStateProvider.java:176)
        at java.base/java.util.Optional.flatMap(Optional.java:289)
        at org.hyperledger.besu.ethereum.bonsai.BonsaiWorldStateProvider.getMutable(BonsaiWorldStateProvider.java:174)
        at org.hyperledger.besu.ethereum.transaction.TransactionSimulator.getWorldState(TransactionSimulator.java:151)
        at org.hyperledger.besu.ethereum.transaction.TransactionSimulator.process(TransactionSimulator.java:131)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.EthCall.resultByBlockHeader(EthCall.java:78)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.AbstractBlockParameterOrBlockHashMethod.latestResult(AbstractBlockParameterOrBlockHashMethod.java:68)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.AbstractBlockParameterOrBlockHashMethod.handleParamTypes(AbstractBlockParameterOrBlockHashMethod.java:98)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.EthCall.response(EthCall.java:102)
        at org.hyperledger.besu.ethereum.api.jsonrpc.execution.BaseJsonRpcProcessor.process(BaseJsonRpcProcessor.java:44)
        at org.hyperledger.besu.ethereum.api.jsonrpc.execution.TracedJsonRpcProcessor.process(TracedJsonRpcProcessor.java:41)
        at org.hyperledger.besu.ethereum.api.jsonrpc.execution.TimedJsonRpcProcessor.process(TimedJsonRpcProcessor.java:45)
        at org.hyperledger.besu.ethereum.api.jsonrpc.execution.JsonRpcExecutor.execute(JsonRpcExecutor.java:92)
        at org.hyperledger.besu.ethereum.api.handlers.AbstractJsonRpcExecutor.executeRequest(AbstractJsonRpcExecutor.java:87)
        at org.hyperledger.besu.ethereum.api.handlers.JsonRpcObjectExecutor.execute(JsonRpcObjectExecutor.java:57)
        at org.hyperledger.besu.ethereum.api.handlers.JsonRpcExecutorHandler.lambda$handler$0(JsonRpcExecutorHandler.java:57)
        at java.base/java.util.Optional.ifPresentOrElse(Optional.java:196)
        at org.hyperledger.besu.ethereum.api.handlers.JsonRpcExecutorHandler.lambda$handler$2(JsonRpcExecutorHandler.java:54)
        at io.vertx.ext.web.impl.BlockingHandlerDecorator.lambda$handle$0(BlockingHandlerDecorator.java:48)
        at io.vertx.core.impl.ContextBase.lambda$null$0(ContextBase.java:137)
        at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:264)
        at io.vertx.core.impl.ContextBase.lambda$executeBlocking$1(ContextBase.java:135)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:833)
2023-06-10 03:54:17.164+00:00 | Timer-0 | INFO  | DNSResolver | Resolved 319 nodes
```

This happens on a new machine, new Rocket Pool install, new sync. It was fully synced and working but throws this error after a couple of hours (around 2-3) and stays there. Restarting it makes it work again until the same error pops up (again, after a couple of hours). This is a default Rocket Pool install. Another note: syncing took about 2.5-3 days rather than the ~6 hours I've read everywhere. Not sure if this is relevant.

Another note: I probably won't be able to do a lot of testing. I've lost quite some time with this already so I'll be moving to another client to get things up and running as soon as possible. I'll try to keep the Besu files if my storage allows it.

### Versions (Add all that apply)
* Software version: `besu/v23.4.1/linux-x86_64/openjdk-java-17`
* Java version: `openjdk version "17.0.7" 2023-04-18`
* OS Name & Version: `Ubuntu 22.04.2 LTS"`
* Kernel Version: `5.14.0-284.11.1.el9_2.x86_64`
* Docker Version: `Docker version 24.0.2, build cb74dfc`
* Consensus Client & Version if using Proof of Stake: `Teku 23.5.0`

### Additional Information (Add any of the following or anything else that may be relevant)
* Besu setup info - genesis file, config options
Standard Rocket Pool install, environment variables as found in the compose file
```
      BESU_JVM_HEAP_SIZE: "0"
      BESU_MAX_BACK_LAYERS: "512"
      CLIENT: besu
      EC_ADDITIONAL_FLAGS: ""
      EC_CACHE_SIZE: ""
      EC_ENGINE_PORT: "8551"
      EC_HTTP_PORT: "8545"
      EC_MAX_PEERS: "50"
      EC_METRICS_PORT: "9105"
      EC_P2P_PORT: "30303"
      EC_WS_PORT: "8546"
      ENABLE_METRICS: "true"
      ETHSTATS_LABEL: ""
      ETHSTATS_LOGIN: ""
      EXTERNAL_IP: 67.211.221.163
      NETWORK: mainnet
      RP_NETHERMIND_ADDITIONAL_MODULES: ""
      RP_NETHERMIND_ADDITIONAL_URLS: ""
      RP_NETHERMIND_COMPLETE_HISTORY: ""
      RP_NETHERMIND_PRUNE_MEM_SIZE: ""
      TTD_OVERRIDE: ""
```
* System info - memory, CPU
Intel(R) Xeon(R) CPU E3-1270 V2 3.5GHz 8-core
29GB RAM

**Comment by matkt** [comment]
What version of Besu are you using?

**Comment by matkt** [comment]
Do you see besu restart in the logs? Maybe besu is killed

**Comment by matkt** [comment]
I see in your ticket that it is after a restart. How do you  stop besu? It seems that Besu was not stopped correctly

**Comment by yorickdowne** [comment]
It’s not OOM, it has enough ram. If after switching to another client you also get errors, it might be a hardware fault. Unusual with xeons but not unheard of. You can check the ipmi on that box for any ECC errors, since I assume this is ECC ram. And check the ssd with smartctl -x 

Good luck!

**Comment by igg2** [comment]
I've got something similar, possibly:
```

**Comment by matkt** [comment]
Did you have version 23.7.1 during the first error (at the time of the restart) or did you update?

**Comment by matkt** [comment]
Has your node finished sync? Where do eth_calls come from

**Comment by igg2** [comment]
Before the restart, the node was synched for several hours and there was no powerfailures. It was frozen with that exception for ~20 minutes. I restarted and it was throwing that exception right away every minute for > 1 hr. Restarted again, still the same thing. I stopped the node and just let it sit for ~1 hour, hoping that it will force a restore of some kind. When I started it again, it threw the same exception again a few times, but also looked like it was trying to resync from peers, which it eventually did and the exceptions went away. They haven't been back for a couple hours now.

Basically, it would be nice if it found a bad place in the chain to try to resync the chain through its peers, rather than just keep throwing the same exception over and over. It's obviously capable of restoring itself, it just seems like it should try to do that from it's peers sooner rather than later.

Also, it shouldn't be throwing exceptions at all really. Whatever happened, the exception came from core Java:
 ```rocketpool_eth1  | 	at java.base/java.util.Optional.get(Optional.java:143)```, which should have been caught earlier in the caller:
 ```rocketpool_eth1  | 	at org.hyperledger.besu.ethereum.bonsai.BonsaiWorldStateProvider.rollMutableStateToBlockHash(BonsaiWorldStateProvider.java:216)
```
which would then be a good opportunity to try to recover that piece of the chain from one of the peers.
Or something, I don't have any familiarity with this code - so I'm just guessing. There should definitately not be exceptions coming from core Java though (again, guessing).
